code/modules/defenses/sentry_flamer.dm

```diff
@@ -99,3 +99,37 @@
 			range_bounds = RECT(x, y - (FLAMER_SENTRY_SNIPER_RANGE/2), FLAMER_SENTRY_SNIPER_RANGE, FLAMER_SENTRY_SNIPER_RANGE)
 
 #undef FLAMER_SENTRY_SNIPER_RANGE
+
+
+/obj/structure/machinery/defenses/sentry/flamer/wy
+	name = "WY 406-FE2 Smart Sentry"
+	icon = 'icons/obj/structures/machinery/defenses/wy_defenses.dmi'
+	desc = "A deployable, fully-automated turret with AI targeting capabilities used by the PMC."
+	health = 300
+	health_max = 300
+	fire_delay = 2 SECONDS
+	disassemble_time = 5 SECONDS
+	hack_time = 25 SECONDS
+	ammo = new /obj/item/ammo_magazine/sentry_flamer/wy
+	sentry_type = "wy_flamer"
+	sentry_range = 6
+	omni_directional = TRUE
+	handheld_type = /obj/item/defenses/handheld/sentry/flamer/wy
+	selected_categories = list(
+		SENTRY_CATEGORY_ROF = ROF_SINGLE,
+		SENTRY_CATEGORY_IFF = FACTION_WEYLAND,
+	)
+
+/obj/structure/machinery/defenses/sentry/flamer/wy/destroyed_action()
+	visible_message("[icon2html(src, viewers(src))] [SPAN_WARNING("The [name] starts spitting out sparks and smoke!")]")
+	playsound(loc, 'sound/mecha/critdestrsyndi.ogg', 25, 1)
+	for(var/i = 1 to 6)
+		setDir(pick(NORTH, EAST, SOUTH, WEST))
+		sleep(2)
+
+	if(ammo.current_rounds != 0)
```

Comment:
```suggestion
	if(ammo.current_rounds)
```

Reply:
It cannot read if you var has a value for some reason when I do this

Reply:
Then just revert it back as it was then, not a big deal.